### DISCUSSION TITLE
Dynamic tasks - how to map inputs if some tasks fail

### DISCUSSION THREAD

**Discussion by kittyandrew** [apache/airflow#34262] [Q&A]
Hi,
I have: task that generates list of data, dynamic tasks that transform that data, and task that combines those two outputs.
The problem is in the fact that sometimes some subtasks of the dynamic tasks can fail, so I added `trigger_rule = "all_done"`. Unfortunately this didn't work due to the "combinator" task (task that requires both inputs).

Since it probably sounds confusing, I created a repro with my exact problem:
```python
from airflow.exceptions import AirflowSkipException
from datetime import datetime, timedelta
from airflow.decorators import task
from airflow import models
from random import randint


@task
def generate_data():
    return list(range(65, 91))


@task
def generate_char(index: int):
    if randint(0, 10) > 7: raise AirflowSkipException
    return chr(index)


@task(trigger_rule="all_done")
def print_results(numbers, chars):
    for n, c in zip(numbers, chars):
        print(n, c) # Do not match, since 'chars' is smaller and list shifted.


with models.DAG(
    "test_job",
    max_active_runs=1,
    default_args={
        "retries": 3,
        "retry_delay": timedelta(minutes=1),
    },
    schedule=None,
    start_date=datetime(2021, 1, 1),
    catchup=False,
) as dag:
    numbers = generate_data()
    chars = generate_char.expand(index=numbers)
    print_results(numbers, chars) # or numbers.zip(chars)
```
This DAG outputs something like:
```
INFO - 65 A
INFO - 66 C
INFO - 67 D
INFO - 68 E
INFO - 69 G
INFO - 70 H
INFO - 71 J
INFO - 72 K
INFO - 73 L
INFO - 74 M
INFO - 75 N
INFO - 76 O
INFO - 77 Q
INFO - 78 T
INFO - 79 U
INFO - 80 X
INFO - 81 Y
INFO - 82 Z
```
As you can see, this just naively zips them together, but what I need is for data in the left list to correspond to outputs on the right.
Which would be like:
```
INFO - 65 A
INFO - 67 C
... etc
```
The simple solution here would be to pass both values to `generate_char`, but I cannot do that, because `generate_char` in my case is a docker container (GKE operator) that runs program I don't have direct control over. 

**Note** that in this simplified example, output is reversible to input (`ord`), but in my actual use case - it's not, think of `hash` function as an example.

I'm using Airflow 2.4.3, but I couldn't find any docs on this or any later version to help with this.

**Comment by sann05** [comment]
PS. I would avoid using randomizers for reproduction examples, but if you replace 
```
if randint(0, 10) > 7: raise AirflowSkipException
```
with 
```
if index % 5 == 1: raise AirflowSkipException
```
You will get the same issue, with the similar output that @kittyandrew received. 

The problem is also actual for:
Version: [v2.7.1](https://pypi.python.org/pypi/apache-airflow/2.7.1)
Git Version: .release:b8c416681c529aad3ef744c193f6e0435c4d0d93

```
65 A
66 C
67 D
68 E
69 F
70 H
71 I
72 J
73 K
74 M
75 N
76 O
77 P
78 R
79 S
80 T
81 U
82 W
83 X
84 Y
85 Z
```

**Reply by kittyandrew** [reply]
Yeah, the randomizer example isn't ideal, thanks for elaborating and verifying against new release.

It seems like what I need here essentially is an option to change behavior of the dynamic task, so that it returns `None` instead of removing the failed value from list (or some analogous behavior to that).
